packages/next/types/global.d.ts

```diff
@@ -13,15 +13,15 @@ declare namespace NodeJS {
 
 declare module '*.module.css' {
   const classes: { readonly [key: string]: string }
-  export default classes
+  export = classes
 }
 
 declare module '*.module.sass' {
   const classes: { readonly [key: string]: string }
-  export default classes
+  export = classes
 }
 
 declare module '*.module.scss' {
   const classes: { readonly [key: string]: string }
-  export default classes
+  export = classes
```

Comment:
By using `typed CSS modules` you can get specific types for each css file. Please look at cli and vscode plugin for better dev experience.

<img width="309" alt="Screen Shot 2020-12-19 at 11 02 05 AM" src="https://user-images.githubusercontent.com/2255993/102680324-d6b05d80-41e9-11eb-8da4-1c987bf5cce7.png">

Reply:
Are the types supplied by Next conflicting with the typed CSS modules (I suppose this is what you're talking about https://github.com/Quramy/typed-css-modules)?

Reply:
@ludofischer  I'm not sure how it will result after your changes will be made, but looks like `{ readonly [key: string]: string }` in your .d.ts will allow all the strings as classNames?

Building specific css module types on save/edit would be preferred way for me, so it will be possible to detect dead CSS code with static analyze tool. BTW is there any for next.js? There is gatsby-plugin-purge-css for this case, which helps to maintain codebase. need something like this for next.js

Reply:
Yeah, the typings allow any string as class name. Building css module types looks like an interesting feature, but I am not familiar enough with the next.js code base to say how it could be implemented. About purgeCSS I just found a guide in the documentation: https://purgecss.com/guides/next.html#customize-postcss-configuration-next-js-9-3

Reply:
@ludofischer `Yeah, the typings allow any string as class name.` For my codebases it is not acceptable, cos I will loose autocomplete and will have a lot of `<div class="undefined" />` in output html in case of typos, or accidentally removed classnames.

Reply:
@ludofischer  Well, in this case there will be no typesafety for used classnames, and a lot of people will get html produced with class="undefined" in case of typo or removed classes from css files. Will complicate code maintenance a lot.

Reply:
I don’t think this pull request makes the situation worse, it’s even a little bit better than before as if this pull request gets merged, you cannot import the wrong class by name.

Reply:
ok, if it correctly infer types from each css file, I'm fine with it.